server/platform/services/cache/redis.go

```diff
@@ -285,3 +306,11 @@ func (r *Redis) GetInvalidateClusterEvent() model.ClusterEvent {
 func (r *Redis) Name() string {
 	return r.name
 }
+
+func sliceMap[S ~[]E, E, R any](slice S, mapper func(E) R) []R {
+	newSlice := make([]R, len(slice))
+	for i, v := range slice {
+		newSlice[i] = mapper(v)
+	}
+	return newSlice
+}
```

Comment:
I was torn on this. Whether to do it or not. The thing is we need to do it for GetMulti, so as to not modify the passed slice from the app layer, otherwise it causes troubles. But I think for Scan, it should be fine. But I leaned towards consistency than these minor gains.

Reply:
Sounds good, consistency is indeed more important than premature optimization. Thanks for clarifying!